e2e-samples/GitIntegration-CommitSelective.ps1

```diff
@@ -0,0 +1,136 @@
+
+$workspaceName = "FILL ME"          # The name of the workspace
+
+$datasetsNames = @("FILL ME")       # The names of the datasets to be committed
+
+$reportsNames = @("FILL ME")        # The name of the reports to be committed
+
+# End Parameters =======================================
+
+function GetFabricHeaders($resourceUrl) {
+
+    #Login to Azure
+    Connect-AzAccount | Out-Null
+
+    # Get authentication
+    $fabricToken = (Get-AzAccessToken -ResourceUrl $resourceUrl).Token
+
+    $fabricHeaders = @{
+        'Content-Type' = "application/json"
+        'Authorization' = "Bearer {0}" -f $fabricToken
+    }
+
+    return $fabricHeaders
+}
+
+function GetWorkspaceByName($baseUrl, $fabricHeaders, $workspaceName) {
+    # Get workspaces    
+    $getWorkspacesUrl = "{0}/workspaces" -f $baseUrl
+    $workspaces = (Invoke-RestMethod -Headers $fabricHeaders -Uri $getWorkspacesUrl -Method GET).value
+
+    # Try to find the workspace by display name
+    $workspace = $workspaces | Where-Object {$_.DisplayName -eq $workspaceName}
+
+    return $workspace
+}
+
+function GetErrorResponse($exception) {
+    # Relevant only for PowerShell Core
+    $errorResponse = $_.ErrorDetails.Message
+
+    if(!$errorResponse) {
+        # This is needed to support Windows PowerShell
+        $result = $exception.Response.GetResponseStream()
+        $reader = New-Object System.IO.StreamReader($result)
+        $reader.BaseStream.Position = 0
+        $reader.DiscardBufferedData()
+        $errorResponse = $reader.ReadToEnd();
+    }
+
+    return $errorResponse
+}
+
+try {
+    # Set up API endpoints
+    $resourceUrl = "https://api.fabric.microsoft.com"
+    $baseUrl = "$resourceUrl/v1"
+
+    $fabricHeaders = GetFabricHeaders $resourceUrl
+
+    $workspace = GetWorkspaceByName $baseUrl $fabricHeaders $workspaceName 
+    
+    # Verify the existence of the requested workspace
+	if(!$workspace) {
+	  Write-Host "A workspace with the requested name was not found." -ForegroundColor Red
+	  return
+	}
+
+    # Check for duplicates in the request items
+    if (($datasetsNames | Group-Object | Where-Object { $_.Count -gt 1 }).Count -gt 0) {
+        Write-Host "Duplicate items found in datasetsNames." -ForegroundColor Red
+        return
+    }
+
+    if (($reportsNames | Group-Object | Where-Object { $_.Count -gt 1 }).Count -gt 0) {
+        Write-Host "Duplicate items found in reportsNames." -ForegroundColor Red
+        return
+    }
+
+    # Get Status
+    Write-Host "Calling GET Status REST API to construct the request body for CommitToGit REST API."
+
+    $gitStatusUrl = "{0}/workspaces/{1}/git/status" -f $baseUrl, $workspace.Id
+    $gitStatusResponse = Invoke-RestMethod -Headers $fabricHeaders -Uri $gitStatusUrl -Method GET
+    
+    # Get selected changes
+    $selectedChanges = @($gitStatusResponse.Changes | Where-Object {
+        ($datasetsNames -contains $_.ItemMetadata.DisplayName -and $_.ItemMetadata.ItemType -eq "dataset") -or 
```

Comment:
Why are we so strict (having just dataset and report)?
I know that ObjectId and LogicalId might be empty/null in some cases (of addition and deletion)
But let's give a generic sample of committing a change for a modified item, using the ObjectId or the name without the limitation of the type.